src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -3062,7 +3058,7 @@ bool TheoryStringsRewriter::checkEntailArithWithEqAssumption(Node assumption,
     return false;
   }
 
-  a = a.substitute(TNode(v), TNode(solution));
+  a = Rewriter::rewrite(a.substitute(TNode(v), TNode(solution)));
```

Comment:
I think this call to rewrite is redundant.  checkEntailArith rewrites a.

Reply:
Done